### ISSUE TITLE
Google Material Design Icons

### ISSUE THREAD

**Issue by kristofzerbe** [fontello/fontello#417]
I have prepared a fontello font with the new material design icons from Google (https://github.com/google/material-design-icons) as described on https://github.com/fontello/fontello/wiki/How-to-suggest-font-to-fontello#how-to-prepare-a-font-for-use-with-fontello. It can be downloaded from: https://drive.google.com/file/d/0B8Fq4dtgh8myY05rT0pJRlBBTGM/view?usp=sharing

It would be great, if they would be integrated into fontello.

Regards
Kristof

**Comment by puzrin** [comment]
LGTM. May be someone wish to verify/extend aliases?

**Comment by puzrin** [comment]
I'd suggest to wait a couple of days, to be sure no updates/fixes needed.

**Comment by puzrin** [comment]
@kristofzerbe , there are some thing to improve, to make this font maintainable in case of updates.

- complete set of icons.
- the same names as for original icons (may be, replacing `_` with `-`).
- the same icon codes as in original font (more easy to make diff in future)
- not sure grouping aliaces like `device` (folder names) a really needed for the search.

https://github.com/google/material-design-icons/tree/master/iconfont - it worth to use font for import to solve all problems at once.

1. Convert TTF to SVG font via FontForge
2. Import SVG font (folltello should load all icons at once, preserving codes and names)
3. Edit names/aliaces via web, or directly `config.json` in text editor.

/cc @shyndman

**Comment by kristofzerbe** [comment]
@puzrin I've used all 48px sized SVG's of the available icons from Google's github download. Some SVG's came up with errors on importing in fontello and I'm not a specialist in SVG, so they are not included in the font I've made.

For the naming I took the original groups (folder name in Google's download), added the orginal file name and replaced _ with -, as you suggested.

The suggestion you've made (convert TTF, Import SVG and edit names) means that you won't use my preliminary work for fontello?

**Comment by puzrin** [comment]
@kristofzerbe , in you preliminary work aliases are useful. Other things need rework (for maintainability). If i use it as is, that will cause a huge time loss on next update to track changes.

**Comment by puzrin** [comment]
> The suggestion you've made (convert TTF, Import SVG and edit names) means that you won't use my preliminary work for fontello?

This is the way to quickly get proper glyph codes (as in original)and to not miss any icon. IMHO, it's the fastest way.

That's also can be edited manually in json file. So, i don't insist, only explain one of possible ways.

**Comment by kristofzerbe** [comment]
Ok, FontForge (for Windows) came up with an error opening the original TTF from https://github.com/google/material-design-icons/tree/master/iconfont. As I said, I'm not a specialist in fonts or SVG.

**Comment by puzrin** [comment]
Wow, something is really broken (checked). I didn't knew it.

Reported bug https://github.com/google/material-design-icons/issues/149

**Comment by puzrin** [comment]
Latest fontforge version open TTF, but SVG output is still broken https://github.com/fontforge/fontforge/issues/2369. Fonsquirell generator also has the same problem.

@kristofzerbe i forgot to say, if you import font and wish to preserve glyph codes, fontello encoding must be set to 'unicode' (via menu -> advanced -> encoding)

**Comment by shyndman** [comment]
Can you guys provide any more information on what's going wrong? Any specific error messages?

**Comment by cm0s** [comment]
Any news about the integration of material design icons?

**Comment by shyndman** [comment]
I now provide an SVG font. Does that help?

https://github.com/google/material-design-icons/blob/master/iconfont/MaterialIcons-Regular.svg

**Comment by cm0s** [comment]
Yes, it definitely helps! Thank you for the link.
However, is there any plan to officially integrate material icons in the fontello set? (probably more a question for @puzrin)

**Comment by puzrin** [comment]
@shyndman yes, that will help a lot to keep everything in sync with your repo.

@cm0s As i sad before - i have no objections against adding GMD icons to fontello. Is someone can add search aliases - that will happen fast, if not - when i have time to do it myself (with updating other fonts).

**Comment by dbarjs** [comment]
It will be great, can i do something to help?

**Comment by dbarjs** [comment]
It will be great, can i do something to help?

**Comment by deepfriedbrain** [comment]
My apologies to comment on an old thread, but is there any update on this?

**Comment by anton-iskandiarov** [comment]
This is quite old actually. Material icons are getting popular, is there any obstacles in adding them? It would be so nice to have all most significant icon fonts in one pack.

**Comment by aldarund** [comment]
Ye, mdi icons would be great..

**Comment by jvlobo** [comment]
are there any updates on this?
4 years have passed since this issue was open and the material design icons are not on fontello yet :(

**Comment by mtomov** [comment]
Would love to have them as direct part as well : ) Thank you

**Comment by LucasArgate** [comment]
Can we restart this issue  to try to resolve? I Readed all issue but i din't understand why it not works:
- Fontello read Font
- Material Design Icons are Font

We have this problem:
"For the naming I took the original groups (folder name in Google's download), added the orginal file name and replaced _ with -, as you suggested." - Writed by @kristofzerbe  

So, Material icons use the content with "_" to selectors in css files right?
Can we create a matrix of selector/alias by players?
Can i pull request with this solution @puzrin ?

Thanks all

**Comment by Ben-CA** [comment]
Would like to see Material Icons incorporated as well.

**Comment by TheDirigible** [comment]
Can someone summarize the blocker for this issue?

**Comment by Ben-CA** [comment]
Would still like to see this: https://fonts.google.com/icons?selected=Material+Icons incorporated.
